### PR TITLE
doc: add a Knowledge Base article about consitency - V2

### PR DESCRIPTION
**Pull request by annastuchlik**
This PR is V2 of https://github.com/scylladb/scylladb/pull/12929 (adding faq question regarding eventual and strong consistency)

It adds a KB article about consistency. The primary goal was to explain the difference between strong and eventual consistency (by @guy9)

### CODE REVIEW

**Review by guy9 — APPROVED** [review]
Thanks @annastuchlik , LGTM

**Comment by annastuchlik** [comment]
@nyh Guy applied your feedback in [V1](https://github.com/scylladb/scylladb/pull/12929). Can this PR be merged now?

File: docs/kb/consistency.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+On the other hand, distributed databases with a BASE model trade off strong consistency with high availability. The records stay available, 
+but once the query has been completed across a majority of `nodes <https://university.scylladb.com/courses/scylla-essentials-overview/lessons/architecture/topic/node/>`_, 
+or as defined by the Consistency Level, the transaction is deemed successful. Data replication across all nodes can take some time, but the data in all 
+the nodes will become consistent eventually. Once the query is deemed successful, queries for the data will consistently provide the updated data, 
+even before data replication reaches the last few nodes.
+
+Early results of eventual consistency data queries may not have the most recent updates. This is because it takes time for updates to reach replicas 
+across a distributed database cluster. In strong consistency, however, data is sent to every replica the moment a query is made. This causes delays 
+because responses to any new requests must wait while the replicas are updated.
```
Comment (nyh) [inline docs/kb/consistency.rst:70]: I think this paragraph is somewhat misleading - it suggests that "strong consistency" needs CL=ALL (wait for all replicas to receive the write), but what about CL=QUORUM for both reads and writes, for example, don't we also consider that strong consistency?

File: docs/kb/consistency.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+across a distributed database cluster. In strong consistency, however, data is sent to every replica the moment a query is made. This causes delays 
+because responses to any new requests must wait while the replicas are updated. 
+
+To summarize, in contrast to SQL’s ACID guarantees, NoSQL databases generally provide BASE guarantees. This enables high availability and relaxes 
+the stringent consistency.
+
+A related feature is `Lightweight Transactions <https://docs.scylladb.com/stable/using-scylla/lwt>`_ (LWT). LWT in ScyllaDB allow the client 
+to modify data based on its current state: that is, to perform an update that is executed only if a row does not exist or contains a certain value. 
+LWT are limited to a single conditional statement, which allows an “atomic compare and set” operation. That is, it checks if a condition is true, 
+and if so, it conducts the transaction. If the condition is not met, the transaction does not go through.
```
Comment (nyh) [inline docs/kb/consistency.rst:78]: Again, this paragraph is 100% true, but it's not clear how it relates to the rest of the text. Are LWT statements "consistent", "strongly consistent", or what?

**Review by nyh — COMMENTED** [review]
I think at this point there is nothing wrong, per se, in this text, but to me it doesn't read like a consistent (pun intended) story. It has a lot of theoretical definitions which at points repeat or even contradict one another, but very little concrete details about Scylla itself except a link to some other "tunable consistency" document. Until now I'm not even sure if you consider writes and reads with CL=QUORUM as "strong consistency".

So I'm not against this text, it's not wrong or anything - I just think it could be better. Either that or I don't understand the target audience.

**Comment by nyh** [comment]
> @nyh Guy applied your feedback in [V1](https://github.com/scylladb/scylladb/pull/12929). Can this PR be merged now?

See my answer above. I think it can be merged in the sense that is not "wrong". But I also don't think I would have found this document very useful as a reader (and I explained in my review above why). We can also merge this and improve it at a later time.

**Comment by guy9** [comment]
Thanks @nyh , please merge it, and as you suggest we can iterate on it in the future. Feel free to edit it if you have the chance. I'm also thinking to turn it into a blog post afterward (it's a highly searched-for topic).

**Comment by nyh** [comment]
Merged.

**Pull request closed without merging**